src/compiler.ts

```diff
@@ -4617,79 +4619,199 @@ export class Compiler extends DiagnosticEmitter {
+            expr = module.select(
+              module.i32(1),
+              module.binary(BinaryOp.EqI32, rightExpr, module.i32(0)),
+              leftExpr
+            );
+            break;
+          }
+          case TypeKind.I8:
+          case TypeKind.U8:
+          case TypeKind.I16:
+          case TypeKind.U16:
+          case TypeKind.I32:
+          case TypeKind.U32: {
+            instance = this.i32PowInstance;
+            if (!instance) {
+              let prototype = this.program.lookupGlobal(CommonNames.ipow32);
+              if (!prototype) {
+                this.error(
+                  DiagnosticCode.Cannot_find_name_0,
+                  expression.range, "ipow32"
+                );
+                expr = module.unreachable();
+                break;
+              }
+              assert(prototype.kind == ElementKind.FUNCTION_PROTOTYPE);
+              this.i32PowInstance = instance = this.resolver.resolveFunction(<FunctionPrototype>prototype, null);
+            }
+            if (!instance || !this.compileFunction(instance)) {
               expr = module.unreachable();
-              break;
+            } else {
+              expr = this.makeCallDirect(instance, [ leftExpr, rightExpr ], expression);
             }
-            let prototype = namespace.members ? namespace.members.get(CommonNames.pow) : null;
-            if (!prototype) {
-              this.error(
-                DiagnosticCode.Cannot_find_name_0,
-                expression.range, "Mathf.pow"
-              );
+            break;
+          }
+          case TypeKind.I64:
+          case TypeKind.U64: {
+            instance = this.i64PowInstance;
+            if (!instance) {
+              let prototype = this.program.lookupGlobal(CommonNames.ipow64);
+              if (!prototype) {
+                this.error(
+                  DiagnosticCode.Cannot_find_name_0,
+                  expression.range, "ipow64"
+                );
+                expr = module.unreachable();
+                break;
+              }
+              assert(prototype.kind == ElementKind.FUNCTION_PROTOTYPE);
+              this.i64PowInstance = instance = this.resolver.resolveFunction(<FunctionPrototype>prototype, null);
+            }
+            if (!instance || !this.compileFunction(instance)) {
               expr = module.unreachable();
-              break;
+            } else {
+              expr = this.makeCallDirect(instance, [ leftExpr, rightExpr ], expression);
             }
-            assert(prototype.kind == ElementKind.FUNCTION_PROTOTYPE);
-            this.f32PowInstance = instance = this.resolver.resolveFunction(<FunctionPrototype>prototype, null);
+            break;
           }
+          case TypeKind.ISIZE:
+          case TypeKind.USIZE: {
+            leftType  = this.currentType;
+            rightExpr = this.compileExpression(right, leftType, Constraints.CONV_IMPLICIT);
+            rightType = this.currentType;
```

Comment:
The types in `leftType` and `rightType` should be set accordingly already, so these three lines can be simplified to just

```ts
// leftType  = this.currentType; // already the case
rightExpr = this.compileExpression(right, rightType, Constraints.CONV_IMPLICIT);
// rightType = this.currentType; // already the case due to conversion ^
```